product-mini/platforms/linux/CMakeLists.txt

```diff
@@ -130,6 +130,23 @@ set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wextra -Wformat -Wformat-security -W
 
 set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Wformat -Wformat-security -Wno-unused")
 
+if (NOT DEFINED SANITIZER)
```

Comment:
maybe `SANITIZER` -> `WAMR_BUILD_SANITIZER`

@wenyongh what's your suggestion?

Reply:
Agree, had better use `WAMR_BUILD_SANITIZER`, and apply these changes to `build-scripts/config_common.cmake`, no need to add similar lines in three files: `product-mini/platforms/linux/CMakeLists.txt`, `samples/wasi-threads/CMakeLists.txt` and `samples-wasm-c-api/CMakeLists.txt`.

Reply:
And this PR adds more sanitizer modes in `compilation_on_android_ubuntu.yml`, which introduces lots of CI jobs (from 515 to 675) and increases the test time when PR is created/merged, my suggestion is to add another CI to daily test them.

Reply:
done